test/e2e/customconfigs/skipjobswithoutqueuename_test.go

```diff
@@ -134,5 +129,66 @@ var _ = ginkgo.Describe("ManageJobsWithoutQueueName", ginkgo.Ordered, func() {
 				}, util.Timeout, util.Interval).Should(gomega.Succeed())
 			})
 		})
+
+		ginkgo.It("should suspend a pod created in the test namespace", func() {
+			var testPod, createdPod *corev1.Pod
+			var podLookupKey types.NamespacedName
+			ginkgo.By("creating a pod without a queue name", func() {
+				testPod = testingpod.MakePod("test-pod", ns.Name).Obj()
+				gomega.Expect(k8sClient.Create(ctx, testPod)).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("verifying that the pod is gated", func() {
+				createdPod = &corev1.Pod{}
+				podLookupKey = types.NamespacedName{Name: testPod.Name, Namespace: ns.Name}
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, podLookupKey, createdPod)).Should(gomega.Succeed())
+					g.Expect(createdPod.Spec.SchedulingGates).Should(gomega.HaveLen(1))
+					g.Expect(createdPod.Spec.SchedulingGates).Should(gomega.BeComparableTo([]corev1.PodSchedulingGate{
+						{Name: "kueue.x-k8s.io/admission"},
+					}))
+					g.Expect(createdPod.ObjectMeta.Labels).To(gomega.BeComparableTo(map[string]string{constants.ManagedByKueueLabel: "true"}))
+					g.Expect(createdPod.Finalizers).Should(gomega.ContainElement(constants.ManagedByKueueLabel))
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("deleting the pod", func() {
+				gomega.Expect(k8sClient.Delete(ctx, testPod)).Should(gomega.Succeed())
+				gomega.Eventually(func(g gomega.Gomega) {
+					err := k8sClient.Get(ctx, podLookupKey, createdPod)
+					g.Expect(apierrors.IsNotFound(err)).Should(gomega.BeTrue())
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
```

Comment:
```suggestion
```

I think if we combine BeforeEach and AfterEach within the Describe block, it should be removed automatically.

Reply:
> the hooks are executed starting from the innermost block and moving outward, so the end result is the same as moving them up to the Describe block.

Yes, this is a problem. We should execute `util.DeleteNamespace(ctx, k8sClient, ns))` first to clear all namespaced objects. After that, we can safely delete `ClusterQueue` and `ResourceFlavor`. Otherwise, we will encounter issues deleting `ClusterQueue` and `ResourceFlavor` because they will still have a finalizer that prevents their deletion.

But I see that it works fine for this test case without this optimization, so I’m okay with it in this PR. We can optimize it in a follow-up PR.